hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -286,6 +287,21 @@ public static Schema getRecordKeySchema() {
     return RECORD_KEY_SCHEMA;
   }
 
+  public static Schema getRecordKeySchema(Option<BaseKeyGenerator> keyGeneratorOpt, Option<SerializableSchema> schemaOpt) {
+    if (keyGeneratorOpt.isPresent()) {
+      List<Schema.Field> schemaFieldList = keyGeneratorOpt.get().getRecordKeyFieldNames().stream()
+          .map(name -> {
+            Schema.Field field = schemaOpt.get().get().getField(name);
+            return new Schema.Field(field.name(), field.schema(), "", field.defaultVal());
+          })
+          .collect(Collectors.toList());
+      Schema recordKeySchema = Schema.createRecord("HoodieRecordKey", "", "", false);
```

Comment:
@alexeykudinkin This works for root-level fields.  Do you think we also need to handle nested fields separately?

Reply:
do we support virtual keys w/ nested fields? if yes we might have to fix. if not, we can take it as a follow up.